public/app/features/explore/utils/ResultProcessor.ts

```diff
@@ -101,7 +101,8 @@ export class ResultProcessor {
       return null;
     }
 
-    const newResults = dataFrameToLogsModel(this.dataFrames, this.intervalMs, this.timeZone);
+    const datasourceId = this.state.datasourceInstance?.meta?.id;
+    const newResults = dataFrameToLogsModel(this.dataFrames, this.intervalMs, this.timeZone, datasourceId);
```

Comment:
Also here, datasourceId is not used in dataFrameToLogsModel so it can be removed.